git push fails; lfs sending to wrong remote repo

I ran `git lfs install` in one of my submodules. I added a bunch of binary files. I then do:

```
git push -f origin
```

Relevant git configuration:

```
[push]
    default = current
[pull]
    rebase = true
[remote]
    pushDefault = fork
```

I have the following remotes (note: `company` is the *redacted* name of the repository organization):

```
$ git remote -v
fork    git@bitbucket.org:rcdailey/core.git (fetch)
fork    git@bitbucket.org:rcdailey/core.git (push)
origin  git@bitbucket.org:company/core.git (fetch)
origin  git@bitbucket.org:company/core.git (push)
```

Note: I'm using SSH protocol for my repository remote URLs.

Result of `git lfs env` is:

```
$ git lfs env
git-lfs/2.7.1 (GitHub; windows amd64; go 1.11.5; git 6b7fb6e3)
git version 2.20.1.windows.1

Endpoint=https://bitbucket.org/company/core.git/info/lfs (auth=none)
  SSH=git@bitbucket.org:company/core.git
Endpoint (pushdefault)=https://bitbucket.org/company/core.git/info/lfs (auth=none)
  SSH=git@bitbucket.org:company/core.git
Endpoint (fork)=https://bitbucket.org/rcdailey/core.git/info/lfs (auth=none)
  SSH=git@bitbucket.org:rcdailey/core.git
LocalWorkingDir=E:\code\frontend\source\Core
LocalGitDir=E:\code\frontend\source\.git\modules\Core
LocalGitStorageDir=E:\code\frontend\source\.git\modules\Core
LocalMediaDir=E:\code\frontend\source\.git\modules\Core\lfs\objects
LocalReferenceDirs=
TempDir=E:\code\frontend\source\.git\modules\Core\lfs\tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=E:\code\frontend\source\.git\modules\Core\lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core
GIT_LFS_PATH=C:\Program Files\Git LFS
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

And the output of my push command, with `GIT_TRACE` enabled:

```
$ GIT_TRACE=1 git push -f origin
10:41:12.121661 exec-cmd.c:236          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
10:41:12.125662 git.c:418               trace: built-in: git push -f origin
10:41:12.134657 run-command.c:643       trace: run_command: unset GIT_DIR GIT_PREFIX; ssh git@bitbucket.org 'git-receive-pack '\''company/core.git'\'''
10:41:13.511567 run-command.c:643       trace: run_command: E:/code/frontend/source/.git/modules/Core/hooks/pre-push origin git@bitbucket.org:company/core.git
10:41:13.585556 exec-cmd.c:236          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
10:41:13.588542 git.c:675               trace: exec: git-lfs pre-push origin git@bitbucket.org:company/core.git
10:41:13.588542 run-command.c:643       trace: run_command: git-lfs pre-push origin git@bitbucket.org:company/core.git
10:41:13.624657 trace git-lfs: exec: git 'version'
10:41:13.667704 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
10:41:13.733841 trace git-lfs: exec: git 'config' '-l'
10:41:13.755816 trace git-lfs: pre-push: refs/heads/feature/lfs 53e51ee3aa4907613875799610b32cf0c6cfe3da refs/heads/feature/lfs 575024471d8a95cd38a79b4b8e3b7229f8f97dc8
10:41:14.789926 trace git-lfs: tq: running as batched queue, batch size of 100
10:41:14.790889 trace git-lfs: run_command: git rev-list --stdin --objects --
10:41:15.659979 trace git-lfs: tq: sending batch of size 100
10:41:15.659979 trace git-lfs: run_command: ssh -- git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload
10:41:16.253958 trace git-lfs: api: batch 100 files
10:41:16.263937 trace git-lfs: HTTP: POST https://bitbucket.org/%7B%7D/%7B60959e6c-fa17-4af4-be62-7066b1ea0b43%7D/info/lfs/objects/batch
10:41:20.155462 trace git-lfs: HTTP: 402
10:41:20.155462 trace git-lfs: HTTP: {"message": "\n********************************************************************************\n[ERROR] Your LFS push failed because you're out of file storage\n[ERROR] Change your plan to get more file storage:\n[ERROR] https://bitbucket.org/account/user/rcdailey/plans/\n********************************************************************************\n    "}
10:41:20.156411 trace git-lfs: api error:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

10:41:20.156411 trace git-lfs: tq: sending batch of size 100
10:41:20.156411 trace git-lfs: ssh cache: git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload
10:41:20.157409 trace git-lfs: api: batch 100 files
10:41:20.158408 trace git-lfs: HTTP: POST https://bitbucket.org/%7B%7D/%7B60959e6c-fa17-4af4-be62-7066b1ea0b43%7D/info/lfs/objects/batch
10:41:23.617935 trace git-lfs: HTTP: 402
10:41:23.617935 trace git-lfs: HTTP: {"message": "\n********************************************************************************\n[ERROR] Your LFS push failed because you're out of file storage\n[ERROR] Change your plan to get more file storage:\n[ERROR] https://bitbucket.org/account/user/rcdailey/plans/\n********************************************************************************\n    "}
10:41:23.617935 trace git-lfs: api error:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

10:41:23.617935 trace git-lfs: tq: sending batch of size 36
10:41:23.617935 trace git-lfs: ssh cache: git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload
10:41:23.618923 trace git-lfs: api: batch 36 files
10:41:23.618923 trace git-lfs: HTTP: POST https://bitbucket.org/%7B%7D/%7B60959e6c-fa17-4af4-be62-7066b1ea0b43%7D/info/lfs/objects/batch
10:41:27.140402 trace git-lfs: HTTP: 402
10:41:27.140402 trace git-lfs: HTTP: {"message": "\n********************************************************************************\n[ERROR] Your LFS push failed because you're out of file storage\n[ERROR] Change your plan to get more file storage:\n[ERROR] https://bitbucket.org/account/user/rcdailey/plans/\n********************************************************************************\n    "}
10:41:27.140402 trace git-lfs: api error:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

Uploading LFS objects:   0% (0/101), 0 B | 0 B/s, done
batch response:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

batch response:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

batch response:
********************************************************************************
[ERROR] Your LFS push failed because you're out of file storage
[ERROR] Change your plan to get more file storage:
[ERROR] https://bitbucket.org/account/user/rcdailey/plans/
********************************************************************************

error: failed to push some refs to 'git@bitbucket.org:company/core.git'
```

One thing that catches my eye about the verbose logs during the push is:

    run_command: ssh -- git@bitbucket.org git-lfs-authenticate rcdailey/core.git upload

Shouldn't it be `company/core.git`??

Basically, it is pushing to the *wrong remote URL*. It should be pushing to `origin`, not `fork`. Why is this broken? I did find [this issue](https://github.com/git-lfs/git-lfs/issues/2915), which seems related, since the `git lfs env` command does not show the `origin` remote (but it does show `fork`). However that is labeled to have been fixed in v2.5 of Git LFS and I'm using 2.7. I'm stuck, help!

--- a/commands/command_pre_push.go
+++ b/commands/command_pre_push.go
@@ -47,7 +47,7 @@ func prePushCommand(cmd *cobra.Command, args []string) {
 	requireGitVersion()
 
 	// Remote is first arg
-	if err := cfg.SetValidRemote(args[0]); err != nil {
+	if err := cfg.SetValidPushRemote(args[0]); err != nil {
 		Exit("Invalid remote name %q: %s", args[0], err)
 	}
 

--- a/t/t-pre-push.sh
+++ b/t/t-pre-push.sh
@@ -1156,3 +1156,28 @@ begin_test "pre-push locks verify 403 with verification unset"
   [ -z "$(git config "lfs.$endpoint.locksverify")" ]
 )
 end_test
+
+begin_test "pre-push with pushDefault and explicit remote"
+(
+  set -e
+  reponame="pre-push-pushdefault-explicit"
+  setup_remote_repo "$reponame"
+  clone_repo "$reponame" "$reponame"
+
+  git remote add wrong "$(repo_endpoint "$GITSERVER" "wrong-url")"
+
+  git config "lfs.$(repo_endpoint "$GITSERVER" "$reponame").locksverify" false
+  git config remote.pushDefault wrong
+  git lfs track "*.dat"
+  echo "hi" > a.dat
+  git add .gitattributes a.dat
+  git commit -m "add a.dat"
+
+  refute_server_object "$reponame" 98ea6e4f216f2fb4b69fff9b3a44842c38686ca685f3f55dc48c5d3fb1107be4 "refs/heads/master"
+
+  GIT_TRACE=1 GIT_TRANSFER_TRACE=1 git push origin master 2>&1 | tee push.log
+
+  assert_server_object "$reponame" 98ea6e4f216f2fb4b69fff9b3a44842c38686ca685f3f55dc48c5d3fb1107be4 "refs/heads/master"
+  ! grep wrong-url push.log
+)
+end_test

